pkg/utils/utils.go

```diff
@@ -212,25 +162,20 @@ func CreateConfiguration(configItem ConfigurationItem) (configuration string) {
 }
 
 func GetSelectedContexts(allContexts []string, configuration string) (selectedContexts []string, err error) {
-	config, err := ParseConfiguration(configuration)
-	if err != nil {
-		return
+	if configuration == "" {
+		return selectedContexts, errors.New("configuration can not be empty")
 	}
 
-	for _, cntxt := range allContexts {
-		contextItem, parseError := ParseContext(cntxt)
-		if parseError != nil {
-			err = parseError
+	configPattern := "^*" + configuration + "$"
+	for _, context := range allContexts {
+		match, regexErr := MatchToPattern(context, configPattern)
```

Comment:
Should this just be `regexp.MatchString(configPattern, context)` ?

Reply:
The change is being made in another clean PR
https://github.com/Open-CMSIS-Pack/cbuild/pull/49